S5JS100: Update the Samsung S5JS100 driver and add the support for RIL cellular.

zxytddd opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->Update the driver code of Samsung S5JS100 for supporting mbed-os 6 and add the support for RIL cellular

### Summary of changes <!-- Required -->

<!-- 
    Please provide the following information: 

    Description of the the change (what is this fixing / adding / removing?).

    Why the change is needed (if this is fixing a reported issue please summarize what
    the issue is and add the reference. E.g. Fixes #17119).

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
    
-->

#### Impact of changes <!-- Optional -->
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

### Documentation <!-- Required -->

<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

facchinm committed:
Arduino_NANO33BLE: don't define Arduino pins if compiling in Arduino core

Commit:
Change Cypress linker script region name from .cy_xip to cy_xip

In some toolchains, in order to use linker symbols referring to the
start and end of a region, the region name must not contain a '.'
character. These changes allow those symbols to be used for the cy_xip
region by renaming it. They also create explicit start and end symbols
for GCC.

Commit:
Create block device for working around a reserved region

Given an underlying block device and the size of the reserved region, a
CyReservedRegionBlockDevice will act as the underlying block device but
operating only outside of the reserved region. It also allows reading
from the reserved region. The reserved region is assumed to start at
address 0 in the underlying block device.

Commit:
Create canonical CyReservedRegionBlockDevice for working with/around Cypress WiFi firmware in external memory

This change allows external memory to be used for other purposes while
the WiFi firmware is stored in it by interacting with it via the
reserved region block device.

Commit:
Create non-public target for Cypress boards with externally-stored WiFi firmware

Targets that inherit from this target will have the defines necessary to
place the WiFi firmware in external storage and enable use of the
external WiFi firmware reserved region block device.
Currently, the only target inheriting from this new target is
CY8CPROTO-062S3-4343W.

zhizhezhu committed:
S5JS100: Update the driver code to ensure it passes the fpga_ci_test_shield test case.

Signed-off-by: zhizhe.zhu <zhizhe.zhu@samsung.com>

zhizhezhu committed:
S5JS100: 1. Add RIL cellular interface 2. Add RIL adaptation code on S5JS100 platform

Signed-off-by: zhizhe.zhu <zhizhe.zhu@samsung.com>

facchinm committed:
STM32H7: dual core: handle interrupts on CM4

zhizhezhu committed:
S5JS100: Fix RTC range.(gcc time_t is signed) Fix RTC does not stop counting after a software reset.

Signed-off-by: zhizhe.zhu <zhizhe.zhu@samsung.com>

zhizhezhu committed:
S5JS100: Fix IAR compile error issues

Signed-off-by: zhizhe.zhu <zhizhe.zhu@samsung.com>

zhizhezhu committed:
S5JS100: Fix ARMC6 compile error issues

Signed-off-by: zhizhe.zhu <zhizhe.zhu@samsung.com>

facchinm committed:
STM32: USBPhy: enable HS clock sleep

facchinm committed:
STM32H747xI: allow overriding SetSysClock

facchinm committed:
Add Arduino Portenta H7 target

facchinm committed:
lwip: use adeguate mem-size for Portenta H7

facchinm committed:
STM32: EMAC: allow compiling on MCUs without DCACHE

facchinm committed:
PortentaH7: add Ethernet support

facchinm committed:
STM32H7: DualCore: allow no forced CM4 boot

facchinm committed:
STM32: QSPI: allow setting SAMPLE_SHIFTING mode

facchinm committed:
Fix missing licenses for Arduino original code

zhizhezhu committed:
S5JS100: Fix mbed-hal rtc test case error with ARMC6 toolchain

Signed-off-by: zhizhe.zhu <zhizhe.zhu@samsung.com>

zhizhezhu committed:
S5JS100: Adjust RIL s5js100 RIL codes, including in the following private repository pr-S5JS100-06_ RIL fix issues.

Signed-off-by: zhizhe.zhu <zhizhe.zhu@samsung.com>

zhizhezhu committed:
S5JS100: Fix integration test case error issues

Signed-off-by: zhizhe.zhu <zhizhe.zhu@samsung.com>

Commit:
Adds basic span implementation

Commit:
Fix endif comment

zhizhezhu committed:
S5JS100: Fix Travis failures issues

Signed-off-by: zhizhe.zhu <zhizhe.zhu@samsung.com>

Commit:
Add iter_reference_t to iterator header

Commit:
Correct span constructors

Commit:
Add container constructor and make_span

Commit:
Prevent warning on subspan<0,N>

Commit:
Fix namespaces and span prototype

Commit:
Remove c++17 inline of constexpr

Commit:
Fix remaining namespaces and add include files

Commit:
Remove explicit constructors

Commit:
Update default initializers

Commit:
Fix newline at end of file

Commit:
Fix pointer and reference operator style

Commit:
Missed a pointer operator

kivaisan commented on connectivity/cellular/source/framework/RIL/RIL_CellularContext.cpp:
```diff
@@ -0,0 +1,1164 @@
+
+    ril_token_t *token1;
+    
+    token1 = ril.send_request(RIL_REQUEST_SET_NETWORK_SELECTION_AUTOMATIC,
```
Why manual PLMN selection has been removed?

zhizhezhu replied:
This PR is based on mbed os 6 to add RIL interface, the pr-S5JS100-06_RIL is based on mbed os 5, so there is no `const char *get_plmn() const` function in the CellularDevice.h
Add our principle in the adaptation process is not to change your commit code.

kivaisan replied:
Manual PLMN was a requested feature by Samsung in the first place.
Current implementation didn't have `get_plmn` API call but it was added during cellular RIL development.

zhizhezhu replied:
OK, I got it and will to add it, thanks

kivaisan commented on connectivity/cellular/source/framework/RIL/RIL_CellularContext.cpp:
```diff
@@ -0,0 +1,1164 @@
+    _data_call_addresses = new data_call_addr_data_t;
+
+    if (response->addresses) {
+        mbed_cellular_util::separate_ip_addresses(response->addresses,
```
These are space delimited lists, so should be
`mbed_cellular_util::separate_space_delimited_addresses(true, response->addresses,`

zhizhezhu replied:
This PR is based on mbed os 6 to add RIL interface, the pr-S5JS100-06_RIL is based on mbed os 5, so there is no `void separate_space_delimited_addresses(bool ipv6,char *orig,char *ip1, size_t ip1_size,char *ip2, size_t ip2_size)` function in the CellularUtil.h
Add our principle in the adaptation process is not to change your commit code.

kivaisan replied:
`separate_space_delimited_addresses` is a new functionality that needs to be added to CellularUtility class. `separate_ip_addresses` does not work because in this case as the input format is incompatible with this one.

It only works if you have single stack but in case of dual stack (where you have multiple IP addresses, `separate_ip_addresses `does not work correctly).

zhizhezhu replied:
OK, I got it and will to add it, thanks

kivaisan commented on connectivity/cellular/source/framework/RIL/RIL_CellularContext.cpp:
```diff
@@ -0,0 +1,1164 @@
+                                                               _data_call_addresses->_local_addr_ipv6,
+                                                               sizeof(_data_call_addresses->_local_addr_ipv6),
+                                                               temp, sizeof(temp));
+        mbed_cellular_util::separate_ip_addresses(response->addresses,
```
`mbed_cellular_util::separate_space_delimited_addresses(false, response->addresses,`

zhizhezhu replied:
the same as above

kivaisan commented on connectivity/cellular/source/framework/RIL/RIL_CellularContext.cpp:
```diff
@@ -0,0 +1,1164 @@
+    }
+
+    if (response->dnses) {
+        mbed_cellular_util::separate_ip_addresses(response->dnses,
```
` mbed_cellular_util::separate_space_delimited_addresses(true, response->dnses,`

zhizhezhu replied:
the same as above

kivaisan commented on connectivity/cellular/source/framework/RIL/RIL_CellularContext.cpp:
```diff
@@ -0,0 +1,1164 @@
+                                                               _data_call_addresses->_secondary_dns_addr_ipv6,
+                                                               sizeof(_data_call_addresses->_secondary_dns_addr_ipv6));
+
+        mbed_cellular_util::separate_ip_addresses(response->dnses,
```
`mbed_cellular_util::separate_space_delimited_addresses(false, response->dnses,`

zhizhezhu replied:
the same as above

kivaisan commented on connectivity/cellular/source/framework/RIL/RIL_CellularContext.cpp:
```diff
@@ -0,0 +1,1164 @@
+    }
+
+    if (response->gateways) {
+        mbed_cellular_util::separate_ip_addresses(response->gateways,
```
`mbed_cellular_util::separate_space_delimited_addresses(true, response->gateways,`

zhizhezhu replied:
the same as above

kivaisan commented on connectivity/cellular/source/framework/RIL/RIL_CellularContext.cpp:
```diff
@@ -0,0 +1,1164 @@
+                                                               _data_call_addresses->_gateway_addr_ipv6,
+                                                               sizeof(_data_call_addresses->_gateway_addr_ipv6),
+                                                               temp, sizeof(temp));
+        mbed_cellular_util::separate_ip_addresses(response->gateways,
```
`mbed_cellular_util::separate_space_delimited_addresses(false, response->gateways,`

zhizhezhu replied:
the same as above

kivaisan commented on connectivity/cellular/source/framework/RIL/RIL_CellularContext.cpp:
```diff
@@ -0,0 +1,1164 @@
+    }
+}
+
+const char *RIL_CellularContext::pdp_type_to_string(pdp_type_t pdp_type)
```
Remove this, as it should be already in `CellularContext`.

zhizhezhu replied:
This PR is based on mbed os 6 to add RIL interface, the pr-S5JS100-06_RIL is based on mbed os 5, so it not in `CellularContext`

kivaisan replied:
Again the same thing. This was not needed earlier, but as it's counterpart (`string_to_pdp_type`) is in `CellularContext`, this should be added there as well.

zhizhezhu replied:
OK, I got it and will to add it, thanks

kivaisan commented on connectivity/drivers/cellular/Samsung/S5JS100_RIL/SAMSUNG_S5JS100_RIL.cpp:
```diff
@@ -0,0 +1,188 @@
+#include "ScopedLock.h"
+
+using namespace mbed;
+typedef enum s5js100_cellular_event_status {
```
This should be moved to `cellular_connection_status_t `list in `CellularCommon.h` even though this is RIL specific. That way we ensure these numbers stay in sync.

zhizhezhu replied:
This PR is based on mbed os 6 to add RIL interface, the pr-S5JS100-06_RIL is based on mbed os 5, so it not in `CellularCommon.h`

kivaisan commented on connectivity/cellular/source/framework/RIL/RIL_CellularContext.cpp:
```diff
@@ -0,0 +1,1164 @@
+#include "LWIPStack.h"
+#include "CellularLog.h"
+#include "ScopedLock.h"
+#include "mbed.h"
```
Including `mbed.h` is allowed only for application code, so replace these (there are multiple these in this PR) with the required headers.

zhizhezhu replied:
This issue has been modified during the resolution of Travis failures, and the code will be submitted later.

Commit:
Use original <span> if present and the right version

pan- commented on targets/TARGET_Samsung/TARGET_SIDK_S5JS100/rtc_api.c:
```diff
@@ -55,7 +54,7 @@ time_t rtc_read(void)
 	long long cnt;
 	cnt = get_counter(); //return aliva countvalue in 40bits
-	t = cnt/32000;		//calculate as seconds
+	t = cnt/31000;		//calculate as seconds
```
This has been changed several time in the PR: `32768` => `32000` => `31000`. Maybe a more descriptive comment would be appropriate.

zhizhezhu replied:
The purpose of these changes is to calibrate the crystal oscillator accuracy of the development board.

pan- commented on targets/TARGET_Samsung/TARGET_SIDK_S5JS100/device/s5js100_systemreset.c:
```diff
@@ -62,13 +62,13 @@
 void up_systemreset(void)
 {
-//#if DEVICE_RTC
+#if DEVICE_RTC
```
Side note: I'm not sure it is expected from the system to output something on the _standard_ output right before reset.

zhizhezhu replied:
When use sortware reset , the system will output `reset..now` log, and use hardware reset, the system has no output.

pan- reviewed (commented):
That's a large PR (maybe it should have been divided in two PRs, one for cellular and the other for the HAL) and I'm not equipped to fully review it. But I have few questions from a high level standpoint: 
- The cellular part of the code seems to rely a lot on the `new` operator. Is there any reason to not use the placement new variant that allows the caller to check if the allocation failed or not ? 
- Is there any documentation available explaining how to use the Radio Interface Layer ? 
- Is it possible to use the Radio Interface Layer outside the S5JS1000 ? 
- Each class of the RIL layer has its own `_mutex_api` and the communication pattern with `lock_and_send_request` is not the easiest to follow. Is there any documentation explaining how deadlock can be prevented as the RIL classes refers and calls into each other ?

zhizhezhu committed:
S5JS100: Fix code review issues

Signed-off-by: zhizhe.zhu <zhizhe.zhu@samsung.com>

rajkan01 committed:
CMake: Create mbed-greentea library

pennam committed:
Portenta: enable ETH power supply

kivaisan reviewed (approved):
LGTM.

Maybe one question; should we have `RIL_CellularDevice::send_at_command` also declared at `CellularDevice` level, like we have in original code? This is a bit questionable, as it is a RIL specific functionality but in order to keep API usage simpler (prevent casting on application layer), it would be easier to have it already on CellularDevice level.

facchinm committed:
Add SPDX-License

zhizhezhu committed:
S5JS100: Refactor the RIL layer adaptation code structure to pass failures CI check

Signed-off-by: zhizhe.zhu <zhizhe.zhu@samsung.com>

facchinm committed:
[Portenta] Add Cmake rules

Completes #13915

hugueskamba committed:
CMake: Add support for Arm's CM3DS_MPS2 target

hugueskamba committed:
CMake: Add support for Arm's MPS2 targets

dustin-crossman committed:
Add component defines for psoc6 boards.

dustin-crossman committed:
Fixup ticker api due to cyhal_timer_t changes.

hugueskamba committed:
CMake: Add mbed-psa library as a dependency to mbed-core if PSA is used

Include PSA source files only if a given Mbed target needs it.
This commit also consolidates CMakeLists.txt files within the platform
directory

hugueskamba committed:
CMake: Add support for ARM's MUSCA B1 target

Add CMakeLists.txt input source file for the target

hugueskamba committed:
CMake: Add support Arm Musca S1 target

Add CMakeLists.txt file for the target

hugueskamba committed:
Add support for all ARM SSG Mbed targets

dustin-crossman committed:
Update Cypress udb-sdio-whd to 1.1.0.19214

dustin-crossman committed:
Update Cypress WHD asset to 1.1.0.19214

dustin-crossman committed:
Update Cypress serial flash asset to 1.0.2.17634

dustin-crossman committed:
Update Cypress psoc6pdl asset to 2.0.0.6211

dustin-crossman committed:
Update Cypress psoc6hal asset to 1.4.0.17634

dustin-crossman committed:
Update Cypress core-lib asset to 1.1.4.17634

dustin-crossman committed:
Update Cypress abstraction-rtos asset to 1.3.0.17634

dustin-crossman committed:
Update Cypress abstraction-resource asset to 1.0.0.17634

zxytddd committed:
S5JS100: Change the location of S5JS100 RIL code

facchinm committed:
STM32H7: FDCAN: use PLL1Q frequency for calculations

jeromecoutant committed:
STM32F2: directory restructuration

Only files move and targets.json updated accordingly

pea-pod committed:
Add SPI bitwidths to ST targets where supported

jeromecoutant committed:
STM32F2: alignment with other families

facchinm committed:
STM32H747_CM4: add RAM_D3 memory section

facchinm committed:
PORTENTA_H7: use default linker/startup scripts

jeromecoutant committed:
STM32F2: ST CUBE drivers update V1.6.0 => V1.9.2

kjansen-adi committed:
Fixed alignment.

kjansen-adi committed:
Added new line at EoF.

jeromecoutant committed:
STM32: RESET GPIO during init is configurable

Default value is disabled

hugueskamba committed:
CMake: Fix Cortex-M33-NS and Cortex-M33 ASM flags

The ASM flags were not supported by armclang, use the armclang alternatives.
Removed armlink --cpu flag as they generate a CMake configuration
time error. Furthermore, it is not necessary to specify the
processor or architecture as the linker auto-detects it from
the input object files.
See: https://developer.arm.com/documentation/101754/0615/armlink-Reference/armlink-Command-line-Options/--cpu-name--armlink-

Commit:
Add SD component and Update SD card SPI interface

To support MAX32630FTHR SD card interface implicitly,
target and SD component configurations updated.

harrisonmutai-arm committed:
Add SPDX identifier

Commit:
Add proper power save mode command format for MTS_DRAGONFLY_L471QG

jeromecoutant committed:
LORA-DRIVER: add pins in json configuration

jeromecoutant committed:
DISCO_L072CZ_LRWAN1: enable LORAWAN by default

harrisonmutai-arm committed:
CMake: add support for Nordic targets

Commit:
Corrected Wi-SUN certificate add and remove functions on Wi-SUN interface

Corrected Wi-SUN certificate add and remove functions to work properly
with .json certificates. .json certificates are used only if certificates
added by add functions are not set. Corrected certificate add and remove
functions to work properly and improved function descriptions.

jeromecoutant committed:
LORA: update test

dustin-crossman committed:
Update Cypress udb-sdio-whd to 1.1.1

rajkan01 committed:
Add CMake support

hugueskamba committed:
CMake: streamline kvstore support

Link KVStore dependency in Mbed OS so application projects do not have to.

dustin-crossman committed:
Fix mode of changed files.

ryanvesdemc committed:
Enabled USBDevice in targets.json for H743ZI2

ryanvesdemc committed:
Enabled USBDEVICE in targets.json for H743ZI2

zxytddd committed:
Merge remote-tracking branch 'arm/master' into HEAD

0xc0170 committed:
Merge pull request #14029 from arduino/can_h7_wrong_frequency

STM32H7: FDCAN: use PLL1Q frequency for calculations

0xc0170 committed:
Merge pull request #14041 from harmut01/add_spdx_core

Add SPDX identifier to files in storage

Commit:
Add internal subfolder to USBDriver's CMakelists

To build USB Drivers for targets which have USBDevice attribute,
subfolder added as header file path.

harrisonmutai-arm committed:
CMake: add cmake support for Toshiba targets

0xc0170 committed:
Merge pull request #14065 from hugueskamba/hk_cmake_streamline_kvstore

CMake: streamline kvstore support

0xc0170 committed:
Merge pull request #14071 from MaximIntegrated/usbdriver-include-fix

Add `internal` subfolder to USBDriver's CMakelists

0xc0170 committed:
Merge pull request #14036 from ahmetalincak/maxim-max32630fthr-SD-configuration

Add SD component and Update SD card SPI interface for Maxim's MAX32630FTHR

Commit:
Maxim targets; Add USB stack implementation

gpsimenos committed:
Rename FLASH_SECTOR_SIZE macro to avoid conflicts

hugueskamba committed:
Correct OSPIF include directory

The include directory was incorrectly named QSPIF.

hugueskamba committed:
CMake: Add support for OSPIF blockdevice

adbridge committed:
Merge pull request #14061 from rajkan01/create_mbed_greentea_framework_lib

CMake: Create mbed-greentea library

hugueskamba committed:
CMake: Add Blockdevice dependency to KVstore library

As KVstore relies on BlockDevice add the dependency
in Mbed OS so applications do not have to specify it

Commit:
Add bare-metal support for MAXIM targets

Commit:
Add SLEEP/DEEPSLEEP support for MAX32630FTHR

rajkan01 committed:
CMake: Add platform greentea test with CMake

rajkan01 committed:
Create Greentea CMake Macro

rajkan01 committed:
Incorporating the review comments

paul-szczepanek-arm committed:
copy authorsation callbacks to gattserver

paul-szczepanek-arm committed:
correct the documentation for GattServer::addService

rajkan01 committed:
Apply suggestions from code review

Co-authored-by: Hugues Kamba <41612201+hugueskamba@users.noreply.github.com>

adbridge committed:
Update events period method to check for invalid values

The period method currently allows any ms value positive, negative
and zero. A negative value means dispatch a non periodic event
ie just once. 0 is unspecified behaviour. This commit forces the
user to use either positive periods or a new constant non_periodic
and should any other value be provided will default to
non_periodic.

A Greentea test case is also provided to check this works as
expected.

zhizhezhu committed:
S5JS100: Support CMake build through adjust code directory structure

Signed-off-by: zhizhe.zhu <zhizhe.zhu@samsung.com>

zhizhezhu committed:
S5JS100: Support CMake build through adjust code directory structure

Signed-off-by: zhizhe.zhu <zhizhe.zhu@samsung.com>

zxytddd committed:
S5JS100: Fix the astyle issue.

zxytddd committed:
S5JS100: Fix the astyle issue.

hugueskamba committed:
CMake: Add support for KVstore Greentea tests

The KVstore Greentea tests can now be built with CMake

hugueskamba committed:
CMake: Add support for Filesystem Greentea tests

The Filesystem Greentea tests can now be built with CMake

hugueskamba committed:
CMake: Add support for Blockdevice Greentea tests

The Blockdevice Greentea tests can now be built with CMake

hugueskamba committed:
CMake: Add support for NFC Greentea test

The NFC Greentea test can now be built with CMake

rajkan01 committed:
CMake: Add CMake support for mbed_hal greentea test

rajkan01 committed:
CMake: Add support for mbed_hal_fpga_ci_test_shield greentea test

rajkan01 committed:
CMake: Add support for mbed_timing_fpga_ci_test_shield greentea test

rajkan01 committed:
CMake: Add CMake support for rtos greentea test

rajkan01 committed:
CMake: Add CMake support for events greentea test

hugueskamba committed:
CMake: Add MbedTLS dependency to LoRaWAN library

This way applications that need LoRaWAN do need to also add MbedTLS

hugueskamba committed:
CMake: Add support for LoRaWAN Greentea test

The LoRaWAN Greentea test can now be built with CMake.

hugueskamba committed:
CMake: Fix path to Ambiq BLE source file

Update the hci_drv_apollo3.c listing as the file location was changed.

rajkan01 committed:
CMake: Add CMake support for netsocket greentea test

hugueskamba committed:
CMake: Add support for netsocket/network Greentea tests

The netsocket/network Greentea tests can now be built with CMake

hugueskamba committed:
CMake: Fix the listing of system_clock.c for STM32F413xH

Each variant now has its own system_clock.c file.
Therefore ensure the correct one is added for each variant.
Also reduce the number of CMakeLists.txt file as each
variant does not have significant number of files.

0xc0170 committed:
Merge pull request #14045 from harmut01/port_nordic_targets

CMake: add support for Nordic targets

0xc0170 committed:
Merge pull request #14079 from hugueskamba/hk_cmake_kvstore_dependency

CMake: Add Blockdevice dependency to KVstore library

paul-szczepanek-arm committed:
remove deprecated examples

0xc0170 committed:
travis: fix 3.5 test bug in attrs

Using specific version (pre 20) should fix the issue https://github.com/ARMmbed/mbed-os/issues/14110

facchinm committed:
cmake: STM32H7: fix mbed-emac target

facchinm committed:
cmake: STM32H7: fix stm32h747 targets

This needs to be defined for "final" targets not contained in their processor subdirectory

facchinm committed:
cmake: STM32H7: enable PORTENTA* target

0xc0170 committed:
Merge pull request #14108 from 0xc0170/travis_remove_py3.5

travis: set attrs to 19.1.0, fixes python 3.5 issue

paul-szczepanek-arm committed:
add missing guards for extended advertising feature

paul-szczepanek-arm committed:
compile out trace when not available

0xc0170 committed:
Merge pull request #14112 from paul-szczepanek-arm/fix-arm-compile

ARM linking issue: compile out trace when not available

0xc0170 committed:
Merge pull request #14074 from harmut01/port_toshiba_targets

CMake: Add cmake support for Toshiba targets

0xc0170 committed:
Merge pull request #14084 from paul-szczepanek-arm/gatt-copy-callbacks

BLE: Fix GattServer callbacks not being copied to GattServer instance of the service

0xc0170 committed:
Merge pull request #14078 from hugueskamba/hk_rename_ospif_include_dir

CMake: Add support for OSPIF blockdevice

0xc0170 committed:
Merge pull request #14111 from paul-szczepanek-arm/ble-fix-ext

BLE: fix scanning/advertising when extended features are available but disabled on host

0xc0170 committed:
Merge pull request #14068 from rvasquez6089/master

Enabled USBDevice in targets.json for H743ZI2

0xc0170 committed:
Merge pull request #13826 from arduino/portenta-mainline

Add Arduino Portenta H7 as target

0xc0170 committed:
Merge pull request #14100 from hugueskamba/hk_cmake_fix_ambiq_ble

CMake: Fix path to Ambiq BLE source file

dustin-crossman committed:
Add back nvram image for CYTFM board.
Add label so nvram image is found for 064B0S2 board.

0xc0170 committed:
Merge pull request #14072 from rajkan01/add_cmake_greentea_platform

CMake: Add CMake to platform greentea test

dustin-crossman committed:
Add missing includes.

dustin-crossman committed:
Addressing review feedback.

dustin-crossman committed:
Ran astyle for mbed code format.

jeromecoutant committed:
STM32F439 : correct cmake compilation

adbridge committed:
Updated event period handling and greentea test

non_periodic constant has been moved out of the Event class and
made static within the events namespace so that it is available
both internally within the class and externally.

The MBED_ASSERT has been changed to MBED_WARNING.

The greentea test has been updated:
1) timings reduced to make the test cases run faster
2) The call handler simplified

adbridge committed:
Fix some whitespace issues

Commit:
Update header file to include SPDX identifier

pan- committed:
BLE: Regenerate CSRK if it is all zeroes.

pan- committed:
BLE: Fix SM random number generation.

Ensure the more than 8 random bytes are fetched from the stack.

rajkan01 committed:
Apply suggestions from code review

Co-authored-by: Hugues Kamba <41612201+hugueskamba@users.noreply.github.com>

rajkan01 committed:
Apply suggestions from code review

Co-authored-by: Hugues Kamba <41612201+hugueskamba@users.noreply.github.com>

gpsimenos committed:
Replace macros

adbridge committed:
Remove unecessary namespace qualified

(Also minor whitespace correction)

adbridge committed:
Fix astyle issues

hugueskamba committed:
STM: Fix heap size formula in scatter files

The heap size was incorrectly calculated.
This fixes it by subtracting the Stack size, any memory chunks allocated
before the start of the application (for vectors and/or crash report), and
finally the size of the application from the total RAM size.

hugueskamba committed:
NXP: Fix heap size formula in scatter files

The heap size was incorrectly calculated.
This fixes it by subtracting the Stack size, any memory chunks allocated
before the start of the application (for vectors and/or crash report), and
finally the size of the application from the total RAM size.

hugueskamba committed:
GigaDevice: Fix heap size formula in scatter files

The heap size was incorrectly calculated.
This fixes it by subtracting the Stack size, any memory chunks allocated
before the start of the application (for vectors and/or crash report), and
finally the size of the application from the total RAM size.

hugueskamba committed:
NRF52840: Fix heap size formula in scatter files

The heap size was incorrectly calculated.
This fixes it by subtracting the Stack size, any memory chunks allocated
before the start of the application (for vectors and/or crash report), and
finally the size of the application from the total RAM size.

hugueskamba committed:
EFM32GG: Fix heap size formula in scatter files

The heap size was incorrectly calculated.
This fixes it by subtracting the Stack size, any memory chunks allocated
before the start of the application (for vectors and/or crash report), and
finally the size of the application from the total RAM size.

hugueskamba committed:
MUSCA_B1: Fix heap size formula in scatter files

The heap size was incorrectly calculated.
This fixes it by subtracting the Stack size, any memory chunks allocated
before the start of the application (for vectors and/or crash report), and
finally the size of the application from the total RAM size.

hugueskamba committed:
Apollo3: Fix heap size formula and stack start address in scatter file

The heapsize was incorrectly calculated.
This fixes it by subtracting the Stack size, any memory chunks allocated
before the start of the application (for vectors and/or crash report), and
finally the size of the application from the total RAM size.

The stack start address should be the top of the RAM which is also fixed.

0xc0170 committed:
Merge pull request #14048 from mikaleppanen/corr_cert_add

Corrected Wi-SUN certificate add and remove functions on Wi-SUN interface

0xc0170 committed:
Merge pull request #14060 from rajkan01/add_cmake_greentea_drivers

CMake: Add CMake to drivers greentea test

0xc0170 committed:
Merge pull request #14066 from jeromecoutant/DEV_F2

STM32F2 update drivers version to CUBE V1.9.2

0xc0170 committed:
Merge pull request #14135 from hugueskamba/hk_heap_size_fix_nxp

NXP: Fix heap size formula in scatter files

0xc0170 committed:
Merge pull request #14129 from hugueskamba/hk_heap_size_fix_silabs

EFM32GG: Fix heap size formula in scatter files

0xc0170 committed:
Merge pull request #14088 from hugueskamba/hk_cmake_greentea_kvstore

CMake: Add support for KVstore Greentea tests

0xc0170 committed:
Merge pull request #14083 from MaximIntegrated/max32630fthr-add-sleep-feature

Add SLEEP/DEEPSLEEP support for MAX32630FTHR

0xc0170 committed:
Merge pull request #14134 from hugueskamba/hk_heap_size_fix_gigadevice

GigaDevice: Fix heap size formula in scatter files

0xc0170 committed:
Merge pull request #14094 from rajkan01/add_cmake_greentea_rtos

CMake: Add CMake support for rtos greentea test

0xc0170 committed:
Merge pull request #14093 from rajkan01/add_cmake_greentea_hal

CMake: Add Support for hal greentea test

Commit:
Use MBED_ALIGN and remove mbed.h from header

0xc0170 committed:
Merge pull request #14095 from rajkan01/add_cmake_greentea_events

CMake: Add CMake support for events greentea test

0xc0170 committed:
Merge pull request #14133 from hugueskamba/hk_heap_size_fix_stm

STM: Fix heap size formula in scatter files

0xc0170 committed:
Merge pull request #14131 from hugueskamba/hk_heap_size_fix_arm_ssg

MUSCA_B1: Fix heap size formula in scatter files

0xc0170 committed:
Merge pull request #14125 from pan-/fix-sm-key-generation

BLE: Fix Security Manager key generation

0xc0170 committed:
Merge pull request #14062 from dustin-crossman/pr/dec_asset_update

Cypress Asset Update

0xc0170 committed:
Merge pull request #14101 from rajkan01/add_cmake_greentea_netsocket

CMake: Add CMake support for netsocket greentea test

0xc0170 committed:
Merge pull request #14130 from hugueskamba/hk_heap_size_fix_nrf52840

NRF52840: Fix heap size formula in scatter files

0xc0170 committed:
Merge pull request #14090 from hugueskamba/hk_cmake_greentea_blockdevice

CMake: Add support for Blockdevice Greentea tests

0xc0170 committed:
Merge pull request #14120 from jeromecoutant/DEV_CMAKE_F4

STM32F439 : correct cmake compilation

0xc0170 committed:
Merge pull request #14103 from hugueskamba/hk_cmake_greentea_netsocket_network

CMake: Add support for netsocket/network Greentea tests

adbridge committed:
Fix whitespace to stop astyle complaining

adbridge committed:
Further whitespace fixes

adbridge committed:
one final whitespace change!

gpsimenos committed:
Remove const suffix

0xc0170 committed:
Merge pull request #14008 from MultiTechSystems/mtqn-cpsms

Add proper power save mode command format for MTS_DRAGONFLY_L471

0xc0170 committed:
Merge pull request #14089 from hugueskamba/hk_cmake_greentea_filesystem

CMake: Add support for Filesystem Greentea tests

hugueskamba committed:
Travis: Remove make.py test for events and littlefs

The removed tests were building applications from the content of their
respective README files. The events test was building with the full profile twice.
The first time with the repository as is; and the second time after deleting
some directories (to simulate the baremetal profile).
The littlefs test was building once with the full profile.

Testing make.py is no longer required as it is no longer supported.
There is already test coverage for APIs in the removed tests.

dustin-crossman committed:
Implement qspi_frequency for Cypress Mbed HAL.

sstaub committed:
Update for Arduino Pinnames

pennam committed:
STM32H7: disable uSD transceiver

Commit:
Corrected nanostack coverity warnings

jeromecoutant committed:
STM32G4: TRNG enabled

+ some cleanup/optimisation in SetSysClock

jeromecoutant committed:
STM32G4: dual bank FLASH detection

dustin-crossman committed:
Added process to reinitialize qspi if qspi_init is called twice without free.

gpsimenos committed:
Use named namespace & improvements

gpsimenos committed:
Use length constant instead of literal

Co-authored-by: Ladislas de Toldi <ladislas@detoldi.me>

0xc0170 committed:
Merge pull request #14109 from paul-szczepanek-arm/master

Remove deprecated connectivity examples

0xc0170 committed:
Merge pull request #14049 from jeromecoutant/PR_L072_LORA

DISCO_L072CZ_LRWAN1: enable LORA by default

0xc0170 committed:
Merge pull request #14104 from hugueskamba/hk_cmake_fix_STM32F413xH_sysclock_listing

CMake: Fix the listing of system_clock.c for STM32F413xH

0xc0170 committed:
Merge pull request #14020 from pea-pod/stm-spi-more-bits

Add SPI bitwidths to ST targets where supported

jeromecoutant committed:
STM32 / CMAKE / mbedtls : correct CMakeLists.txt files

jeromecoutant committed:
STM32 / CMAKE / targets : correct CMakeLists.txt files

jeromecoutant committed:
CMAKE / M33 : DOMAIN_NS issue

0xc0170 committed:
Merge pull request #14032 from jeromecoutant/PR_RESET_GPIO

STM32: RESET GPIO during init is configurable

Commit:
Replace classes with structs where applicable

Commit:
Fix minor style issues

Commit:
Remove nonstandard as_bytes

Commit:
Add deduction guides

Commit:
Add some comment blocks

0xc0170 committed:
Merge pull request #14145 from arduino/config_sd_transceiver

STM32H7: disable uSD transceiver

0xc0170 committed:
Merge pull request #14143 from hugueskamba/hk_ci_remove_extended_tests

Travis: Remove make.py test for events and littlefs

Commit:
Add to list of backports

Commit:
Fix missing defintion

Commit:
Backport dynamic extent when including <span>

Keep make_span functions even when <span> is available for compatibility

Commit:
Fix last extra space issues

adbridge committed:
Revert "one final whitespace change!"

This reverts commit c9a3375042a09e0dad03920806644209fa606e8b.

Commit:
Fix make_span text in comment block

adbridge committed:
Minor formatting tweak to fix file permissions

0xc0170 committed:
Merge pull request #14087 from adbridge/events

Update events period method to check for invalid values

0xc0170 committed:
Merge pull request #14149 from jeromecoutant/PR_G4

STM32G4: enable TRNG and correct dual bank flash detection

0xc0170 committed:
Merge pull request #14166 from arduino/pin_arduino_nano33ble

ARDUINO_NANO33BLE: don't define Arduino pins if compiling in Arduino core

0xc0170 committed:
Merge pull request #14154 from jeromecoutant/PR_STM32_CMAKE

STM32: correct cmake compilation issues

0xc0170 committed:
Merge pull request #14091 from hugueskamba/hk_cmake_greentea_nfc

CMake: Add support for NFC Greentea test

0xc0170 committed:
Merge pull request #14076 from gpsimenos/gp-mbed-tester

Replace macros in MbedTester to avoid conflicts

0xc0170 committed:
Merge pull request #14075 from MaximIntegrated/maxim-usb-stack-dev

Maxim targets; Add USB stack implementation

0xc0170 committed:
Merge pull request #13881 from marcemmers/add-mstd-span

Add std::span implementation for cxxsupport

paul-szczepanek-arm committed:
Update cert for testing server

paul-szczepanek-arm committed:
astyle

0xc0170 committed:
Merge pull request #14173 from paul-szczepanek-arm/patch-3

Update cert for testing server

0xc0170 committed:
Merge pull request #13968 from hugueskamba/hk_cmake_arm_ssg_support

CMake: Add support for all ARM SSG Mbed targets

0xc0170 committed:
Merge pull request #14155 from jeromecoutant/PR_M33_CMAKE

CMAKE / M33 : DOMAIN_NS issue

0xc0170 committed:
Merge pull request #14148 from mikaleppanen/nano_coverity_corr_master

Corrected nanostack coverity warnings

0xc0170 committed:
Merge pull request #14080 from MaximIntegrated/max32630fthr-bare-metal-support

Add bare-metal support for MAXIM targets

0xc0170 committed:
Merge pull request #14063 from dustin-crossman/pr/wifi_reserved_region

Support arbitrary storage alongside WiFi firmware in external storage on Cypress targets

0xc0170 committed:
Merge pull request #14023 from kylejansen/sdpk1-qspi

Adding QSPI support to SDP-K1 board support package.

0xc0170 committed:
Merge pull request #14098 from hugueskamba/hk_cmake_greentea_lorawan

CMake: Add support for LoRaWAN Greentea tests

0xc0170 committed:
Merge pull request #14144 from sstaub/STM32-Arduino-Pinnames

Update for Arduino Pinnames

0xc0170 committed:
Merge pull request #14132 from hugueskamba/hk_heap_size_fix_apollo3

Apollo3: Fix heap size formula and stack start address in scatter file

HarukaMa committed:
Merge branch 'master' of https://github.com/zxytddd/mbed-os
